### PR TITLE
Import base nessus policy

### PR DESCRIPTION
**Pull request by KyleEvers**

### CODE REVIEW

**Review by dav3r — DISMISSED** [review]
I think `import_base_policy()` should be broken up into two functions:
- `upload_file()`
- `import_policy()`

Call each function in `main()`, followed by an appropriate `assert` statement to confirm that the results are as expected (similar to the pattern in `nessus6.py`).  That way, you can (somewhat) eliminate the special case you have up in `__make_request()` for `if target == POLICY_UPLOAD`.  You'll have to move some of that logic into the `elif method == 'POST'` conditional, but it will be cleaner than now.

File: ansible/roles/nessus/files/nessus_base.py [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+import copy
+import glob
+import time
+import logging
+import ssl
+import json
+import requests
+import copy
+import os
+
```
Comment (jsf9k) [inline ansible/roles/nessus/files/nessus_base.py]: I personally like imports to be in alphabetical order, so I can quickly find the one I'm looking for or determine that one is missing.

**Review by jsf9k — COMMENTED** [review]
I don't have any other comments, other than those that @dav3r mentioned in his review.

File: ansible/roles/nessus/files/nessus_base.py [hunk 2]
```diff
@@ -80,19 +80,24 @@ def __make_request(self, target, method, payload=None):
...
-                headers = {}
-                headers['X-Cookie'] = 'token={!s}'.format(self.token)
-                files = {'Filedata': (BASE_POLICY_FILE_NAME, open(BASE_POLICY_FILE_NAME, 'rb'), 'text/xml')}
-                response = requests.post(self.url + target, headers=headers, files=files, verify=VERIFY_SSL)
-                LOGGER.info('Policy Uploaded to Nessus Server')
-                resp_dict = json.loads(response.text)
-                response = requests.post(self.url + POLICY_IMPORT, headers=headers, data={'file':resp_dict['fileuploaded']}, verify=VERIFY_SSL)
-                LOGGER.info('Base Policy Imported to Nessus Server Policies')
-            elif method == 'GET':
+            # if target == FILE_UPLOAD:
```
Comment (jsf9k) [inline ansible/roles/nessus/files/nessus_base.py]: Remove this stuff if it's not needed.  There's no need to leave dead code around that can be retrieved from the git history.

File: ansible/roles/nessus/files/nessus_base.py (hunk 1 above)
Comment (jsf9k) [reply ansible/roles/nessus/files/nessus_base.py]: If only OCD were the extent of my problems, @KyleEvers!

File: ansible/roles/nessus/files/nessus_base.py [hunk 3]
```diff
@@ -83,7 +83,8 @@ def __make_request(self, target, method, payload=None, files=None):
                 response = requests.get(self.url + target, headers=headers, params=payload, verify=VERIFY_SSL)
             elif method == 'POST':
                 if files:
-                    headers = ['X-Cookie'] = 'token={!s}'.format(self.token)
+                    # This reassigning of headers is to remove the content type assignment in line 71
```
Comment (dav3r) [inline ansible/roles/nessus/files/nessus_base.py]: So you think line 71 is always going to be line 71, huh?  That's adorable.

Comment (KyleEvers) [reply ansible/roles/nessus/files/nessus_base.py]: No good deed goes unpunished!

File: ansible/roles/nessus/files/nessus_base.py [hunk 4]
```diff
@@ -84,8 +84,7 @@ def __make_request(self, target, method, payload=None, files=None):
                 response = requests.get(self.url + target, headers=headers, params=payload, verify=VERIFY_SSL)
             elif method == 'POST':
                 if files:
-                    headers = {}
-                    headers['X-Cookie'] = 'token={!s}'.format(self.token)
+                    headers = ['X-Cookie'] = 'token={!s}'.format(self.token)
```
Comment (dav3r) [inline ansible/roles/nessus/files/nessus_base.py]: I don't think this line is valid.  Shouldn't it be:
`headers = {'X-Cookie': 'token={!s}'.format(self.token)}`

**Review by dav3r — APPROVED** [review]
Looks great now, good work @KyleEvers

**Pull request merged**